compiler/resolution/lowerIterators.cpp

```diff
@@ -1532,6 +1533,9 @@ static void markLoopProperties(ForLoop* forLoop, BlockStmt* ibody,
 
           loop->setAdditionalLLVMMetadata(llvmAttrs);
         }
+        loop = LoopStmt::findEnclosingLoop(loop->prev);
+        if (loop && !ibody->contains(loop))
```

Comment:
I guess I wonder when this `if` would ever evaluate as `true`. I would have expected `findEnclosingLoop` would guarantee it.

Reply:
Hmm.. good question. Do we flatten functions before iterator lowering? Spitballing

```chpl
proc foo() {
  for ...  // this can be the "enclosing" loop
    iter bar() {
      for ...  // this is the loop we're considering
        yield ...;
    }
    for x in bar() ...
}
```

May be science-fiction, but I don't think `findEnclosingLoop` itself covers this case by looking at its implementation. The question is whether this AST can appear during iterator lowering.

I can try.

Reply:
I think it doesn't have to be sci-fi:

.We inline iterators from inside out in loop nests. Meaning when you inline an inner loop, its `yield` (while still in the AST) will have the outer loop as its enclosing loop. I am getting this trigger for some simple String, code, too. So, I am keeping what I have as is.